samples/dlp_inspect_image_file.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2019, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId = 'YOUR_PROJECT_ID', filepath = 'path/to/image.png') {
+  // [START dlp_inspect_image_file]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const filepath = 'path/to/image.png';
+
+  // Imports the Google Cloud Data Loss Prevention library
+  const dlp = require('@google-cloud/dlp');
+
+  // Import other required libraries
+  const fs = require('fs');
+  const util = require('util');
+  const readFile = util.promisify(fs.readFile);
+
+  // Instantiates a client
+  const client = new dlp.DlpServiceClient();
+
+  async function inspectImageFile() {
+    // Get the bytes of the file
+    const fileBytes = Buffer.from(await readFile(filepath)).toString('base64');
+
+    // Construct request
+    const request = {
+      parent: client.projectPath(projectId),
+      item: {
+        byteItem: {
+          type: 'IMAGE',
+          data: fileBytes,
+        },
+      },
+      inspectConfig: {
+        // The infoTypes of information to match
+        infoTypes: [
+          {name: 'PHONE_NUMBER'},
+          {name: 'EMAIL_ADDRESS'},
+          {name: 'CREDIT_CARD_NUMBER'},
+        ],
+        // Whether to include the matching string
+        includeQuote: true,
+      },
+    };
+
+    // Run request
+    const [response] = await client.inspectContent(request);
+    const findings = response.result.findings;
+    console.log(`Findings: ${!findings.length ? 'None' : ''}`);
+    for (const finding of findings) {
+      console.log(`  Quote: ${finding.quote}`);
+      console.log(`  Info type: ${finding.infoType.name}`);
+      console.log(`  Likelihood: ${finding.likelihood}`);
+    }
+  }
+
+  inspectImageFile(projectId, filepath);
+  // [END dlp_inspect_image_file]
+}
+
+main(...process.argv.slice(2));
```

Comment:
there's been discussion in the [Node.js tooling working group](https://github.com/nodejs/tooling/issues/19) about providing a slightly higher level abstraction on process.argv; potentially Node.js would provide an object.

I agree with the decision to drop yargs so that folks don't have a dependency, but I've definitely found it can be a bit confusing for folks as to why they need to perform a slice operation.

Reply:
Yeah, I made this as simple as possible here while still allowing the sample to be runnable from the command-line. Users on cloud.google.com won't see the slice, they'll just be copying the code between the START and END blocks. I imagine @beccasaurus and co. might auto-generate a more interesting/helpful CLI wrapper.

Reply:
Yep, we use yargs with `--input_param="value"` flags for all sample input parameters